deltacat/compute/compactor/steps/repartition.py

```diff
@@ -0,0 +1,132 @@
+import importlib
+import logging
+from contextlib import nullcontext
+import pyarrow.compute as pc
+import pyarrow as pa
+from typing import List, Optional
+from deltacat.types.media import StorageType, ContentType
+import ray
+from deltacat import logs
+from deltacat.compute.compactor import DeltaAnnotated
+from deltacat.compute.compactor.model.repartition_result import RePartitionResult
+from deltacat.storage import interface as unimplemented_deltacat_storage
+from deltacat.storage import Partition
+from deltacat.utils.ray_utils.runtime import (
+    get_current_ray_task_id,
+    get_current_ray_worker_id,
+)
+from deltacat.utils.common import ReadKwargsProvider
+from deltacat.utils.performance import timed_invocation
+from deltacat.utils.metrics import emit_timer_metrics, MetricsConfig
+from deltacat.storage import Delta
+
+if importlib.util.find_spec("memray"):
+    import memray
+
+logger = logs.configure_deltacat_logger(logging.getLogger(__name__))
+
+
+def _timed_repartition(
+    annotated_delta: DeltaAnnotated,
+    repartition_type: str,
+    repartition_args: dict,
+    destination_partition: Partition,
+    enable_profiler: bool,
+    read_kwargs_provider: Optional[ReadKwargsProvider],
+    deltacat_storage=unimplemented_deltacat_storage,
+) -> RePartitionResult:
+
+    if repartition_type == "range":
```

Comment:
Could we make repartition_type as an enum to clearly show what kind of repartition types are supported?